c10/core/Layout.h

```diff
@@ -6,7 +6,7 @@
 #include <iostream>
 
 namespace c10 {
-enum class Layout : int8_t { Strided, Sparse, Mkldnn };
+enum class Layout : int8_t { Strided, Sparse, Mkldnn, Unknown };
```

Comment:
why do you need this?

Reply:
https://github.com/pytorch/pytorch/blob/gh/izdeby/33/head/aten/src/ATen/native/sparse/SparseTensor.cpp#L162
Avoid crash in case when Layout was not initialized.

Reply:
don't do this.  This essentially makes every `Layout` an `optional<Layout>` and every usage has to check it's actually set.